From 3ba1064d95e823d035ba217cf182c1b460bb065b Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Tue, 24 Apr 2018 15:56:45 +0300
Subject: [PATCH 138/706] enetc: Clear TXF IDR bits too, for future possible
 use of FI TxBD field

Make sure all Tx events get cleared, even if FI is not currenlty used.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit dea12eb69edb467ea220ab86987572813746e060)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc.c b/drivers/net/ethernet/freescale/enetc/enetc.c
index 6ab0253d9659..4e463b10eb5a 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc.c
@@ -339,7 +339,8 @@ static int enetc_clean_tx_ring(struct enetc_bdr *tx_ring)
 		if (is_eof) {
 			tx_frm_cnt++;
 			/* re-arm interrupt source */
-			enetc_wr_reg(tx_ring->idr, BIT(tx_ring->index));
+			enetc_wr_reg(tx_ring->idr, BIT(tx_ring->index) |
+				     BIT(16 + tx_ring->index));
 		}
 
 		if (unlikely(!bds_to_clean))
-- 
2.17.1

